src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs

```diff
@@ -1353,15 +1353,43 @@ public static string GetConfusionTable(IHost host, IDataView confusionDataView,
             host.CheckValue(confusionDataView, nameof(confusionDataView));
             host.CheckParam(sample == -1 || sample >= 2, nameof(sample), "Should be -1 to indicate no sampling, or at least 2");
 
-            // Get the class names.
-            int countCol;
-            host.Check(confusionDataView.Schema.TryGetColumnIndex(MetricKinds.ColumnNames.Count, out countCol), "Did not find the count column");
-            var type = confusionDataView.Schema[countCol].Annotations.Schema.GetColumnOrNull(AnnotationUtils.Kinds.SlotNames)?.Type as VectorDataViewType;
-            host.Check(type != null && type.IsKnownSize && type.ItemType is TextDataViewType, "The Count column does not have a text vector metadata of kind SlotNames.");
+            var weightColumn = confusionDataView.Schema.GetColumnOrNull(MetricKinds.ColumnNames.Weight);
+            bool isWeighted = weightColumn.HasValue;
 
+            var confusionMatrix = GetConfusionTableAsType(host, confusionDataView, binary, sample, false);
+            var confusionTableString = GetConfusionTableAsString(confusionMatrix, false);
+
+            // if there is a Weight column, return the weighted confusionMatrix as well, from this function.
+            if (isWeighted)
+            {
+                confusionMatrix = GetConfusionTableAsType(host, confusionDataView, binary, sample, false);
```

Comment:
>confusionMatrix = GetConfusionTableAsType(host, confusionDataView, binary, sample, false); [](start = 16, length = 90)

why we doing this again?
We already did that in line 1359

Reply:
it should be true instead of false.

---
In reply to: [275079866](https://github.com/dotnet/machinelearning/pull/3250#discussion_r275079866) [](ancestors = 275079866)

Reply:
thanks, i move the Weight inside the method, to avoid doing this twice, than forgot to update here. 

---
In reply to: [275080596](https://github.com/dotnet/machinelearning/pull/3250#discussion_r275080596) [](ancestors = 275080596,275079866)